src/cider/nrepl/middleware/refresh.clj

```diff
@@ -107,8 +116,10 @@
   (let [{:keys [exec]} (meta session)]
     (exec id
           (fn []
-            (locking refresh-tracker
-              (vreset! refresh-tracker (track/tracker))))
+            ;; This used to be a `locking`-based call to our `clear` implementation
+            ;; Now it merely "enqueues" a clearing, because the "refresh-clear" op is called synchronously by CIDER,
```

Comment:
It's best to avoid CIDER specifics here, as the middleware aims to be client-neutral outside of Emacs-specific features like Eldoc.